commands/show.go

```
@@ -50,16 +50,16 @@ func runShow(env *Env, opts showOptions, args []string) error {
 
 	snap := b.Snapshot()
 
-	if len(snap.Comments) == 0 {
+	if len(snap.Comments()) == 0 {
```

This kind of check should never be required!
It would be better if b.Snapshot() would fail early in construction to prevent such error checks, which btw. could easily be forgotten.

---

Idea:
Separate comments from a snapshot. E.g. A bug only holds the title, labels, author (reference) and a description.
Important, the description is not treated as a comment! As a bug might have an *empty* description but never *no* description.
Then a snapshots holds a reference to a *Conversation*, which will then hold all comments. So a conversation is a separate aggregate root from a bug. When a bug changes, it publishes a TimelineItem to the Conversation.

With this design a bug can exists without Comments and above check won't be necessary anymore.
Also a bug could be loaded without loading the whole conversation and a conversation could be used to provide an RSS feed.

---

Resolved with commit 6f21fb4c42305d8167b0043be3b641686c08ca42